iroh-rpc-types/src/gateway.rs

```diff
@@ -1,10 +1,23 @@
 use derive_more::{From, TryInto};
-use quic_rpc::{message::RpcMsg, Service};
+use quic_rpc::{
+    message::{Msg, RpcMsg, ServerStreaming},
+    Service,
+};
 use serde::{Deserialize, Serialize};
 
+use crate::RpcResult;
+
 /// Gateway address
 pub type GatewayAddr = crate::addr::Addr<GatewayService>;
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct WatchRequest;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct WatchResponse {
```

Comment:
Just FYI while the request has to be unique, the response does not have to. So in theory you could have the WatchRequest return VersionResponses.

Probably not a good idea though, since we might want to add additional info to the WatchResponse.

But one thing we could do: VersionRequest and WatchRequest / Response are common to all 3 services, and we want them to contain the same info for all 3 services. So we could move Watch/Version Request/Response to the top level of iroh-rpc-types. I think that would be a nice bit of DRY.